experimental/skills/DialogToDialog/DialogRootBot/Bots/RootBot.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+
+namespace Microsoft.BotBuilderSamples.DialogRootBot.Bots
+{
+    public class RootBot<T> : ActivityHandler
+        where T : Dialog
+    {
+        private readonly ConversationState _conversationState;
+        private readonly Dialog _mainDialog;
+
+        public RootBot(ConversationState conversationState, T mainDialog)
+        {
+            _conversationState = conversationState;
+            _mainDialog = mainDialog;
+        }
+
+        public override async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken = default)
+        {
+            if (turnContext.Activity.Type != ActivityTypes.ConversationUpdate)
+            {
+                // Run the Dialog with the Activity.
+                await _mainDialog.RunAsync(turnContext, _conversationState.CreateProperty<DialogState>("DialogState"), cancellationToken);
+            }
+            else
+            {
+                await base.OnTurnAsync(turnContext, cancellationToken);
+            }
+
+            // Save any state changes that might have occured during the turn.
```

Comment:
```suggestion
            // Save any state changes that might have occurred during the turn.
```

Reply:
Updated comment

---
In reply to: [356195078](https://github.com/microsoft/BotBuilder-Samples/pull/2032#discussion_r356195078) [](ancestors = 356195078)